docs/ingress/virtualserver.md

```diff
@@ -0,0 +1,325 @@
+
+In this step, we will add TLS encryption to the Brewz VirtualServer resource.
+
+### Create a certificate
+Since you are running this lab in a closed ecosystem (UDF), you do not have the ability to easily create external DNS records and obtain a public TLS certificate; we will look at this ability in another lab leveraging F5 Distributed Cloud.  Due to this limitation, we will create a self-signed certificate.
+
+Run the following commands via SSH on the K3s server using the *SSH* or *Web Shell* UDF Access Methods:
+```bash
+# Create the key and cert
+sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/brewz-selfsigned.key -out /etc/ssl/certs/brewz-selfsigned.crt
+```
+
+OpenSSL will prompt you with questions about your cert request:
+- Country Name: US
+- State or Province Name: WA
+- Locality Name: Seattle
+- Organization Name: F5
+- Organizational Unit Name: Brewz
+- Common Name: brewz.f5demo.com
+- Email Address: brewsz@f5demo.com
+
+### Create K8s Secret for the Cert and Key
+Now that we have a self-signed certificate, we need to add it to our K8s cluster as a secret.
+
+Run the following commands via SSH on the K3s server using the *SSH* or *Web Shell* UDF Access Methods:
+
+```shell
+sudo kubectl create secret tls brewz-tls --key=/etc/ssl/private/brewz-selfsigned.key --cert=/etc/ssl/certs/brewz-selfsigned.crt
+```
+
+Now that your secret is created, let's take a look at it.
+
+Run the following command from your laptop:
+```shell
+kubectl describe secret brewz-tls -n nginx-ingress
+```
+
+Your output should look simiar to:
+```shell
+Name:         brewz-tls
+Namespace:    nginx-ingress
+Labels:       <none>
+Annotations:  <none>
+
+Type:  kubernetes.io/tls
+
+Data
+====
+tls.crt:  1164 bytes
+tls.key:  1704 bytes
+```
+
+### Modify VirtualServer Resource
+The final step is to update our Brewz VirtualServer resource to leverage the new TLS certificate.
+
+In VSCode, open the */manifests/brewz/virtual-server.yml* file and add the following fields to the virtualserver:
+
+```yaml
+secret: brewz-tls
+redirect:
+  enable: true
+```
+
+The final file should look like the example below:
+```yaml
+apiVersion: k8s.nginx.org/v1
+kind: VirtualServer
+metadata:
+  name: brewz
+spec:
+  host: brewz.f5demo.com
+  tls:
+    secret: brewz-tls
+  upstreams:
+    - name: spa
+      service: spa
+      port: 80
+    - name: api
+      service: api
+      port: 8000
+      healthCheck:
+        enable: true
+        path: /api/stats
+        interval: 20s
+        jitter: 3s
+        port: 8000
+  routes:
+    - path: /
+      action:
+        pass: spa
+    - path: /api
+      action:
+        pass: api
+      errorPages:
+        - codes: [404]
+          return:
+            code: 404
+            type: application/json
+            body: |
+              {\"msg\": \"Could not find the product!\"}
+            headers:
+            - name: x-debug-original-statues
```

Comment:
Not familiar with this header, but is there a typo in it?

Reply:
yes, typo.  This header is from the NGINX docs so I included it in our demo